packages/graphql/src/schema/validation/custom-rules/utils/same-type-argument-as-field.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { EnumTypeDefinitionNode, ArgumentNode, FieldDefinitionNode, ValueNode } from "graphql";
+import { Kind } from "graphql";
+import { fromValueKind, getInnerTypeName, isArrayType } from "./utils";
+import { isSpatial, isTemporal } from "../../../../constants";
+import { DocumentValidationError } from "./document-validation-error";
+
+export function sameTypeArgumentAsField({
```

Comment:
`assertArgumentHasSameTypeAsField`? Struggling with the current naming as to what this is meant to do.